src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -590,4 +601,8 @@ private void sendGroupKeyRequest(String streamId, String publisherId, Date start
         StreamMessage request = msgCreationUtil.createGroupKeyRequest(publisherId, streamId, encryptionUtil.getPublicKeyAsPemString(), start, end);
         publish(request);
     }
+
+    private String newRequestId() {
+        return UUID.randomUUID().toString();
```

Comment:
This works of course, but wondering if adding some human-readable part could help with debugging e.g. an incrementing number so it's easy to see the sequence in which the ids were generated, that's definitely helped me in debugging the JS client.

Reply:
Hmm, could be!